src/repl.ml

```diff
@@ -129,23 +129,40 @@ let create_terminal instance sandbox =
         Extension_instance.set_repl instance term;
         Ok term))
 
-let get_code text_editor =
-  let selection = Vscode.TextEditor.selection text_editor in
-  let start_line = Vscode.Selection.start selection |> Vscode.Position.line in
-  let end_line = Vscode.Selection.end_ selection |> Vscode.Position.line in
-  let start_char =
-    Vscode.Selection.start selection |> Vscode.Position.character
-  in
-  let end_char = Vscode.Selection.end_ selection |> Vscode.Position.character in
-  let document = Vscode.TextEditor.document text_editor in
-  if start_line = end_line && start_char = end_char then
-    let line = Vscode.TextDocument.lineAt document ~line:start_line in
-    Vscode.TextLine.text line
+let get_selected_code text_editor =
+  let selection = TextEditor.selection text_editor in
+  let document = TextEditor.document text_editor in
+  if Selection.isEmpty selection then None
   else
-    Vscode.TextDocument.getText document ~range:(selection :> Vscode.Range.t) ()
+    let selected_text =
+      TextDocument.getText document ~range:(selection :> Range.t) ()
+    in
+    Some selected_text
+
+let get_uri text_editor =
+  text_editor |> Vscode.TextEditor.document |> Vscode.TextDocument.uri
+  |> Vscode.Uri.fsPath
 
-let prepare_code code =
-  if String.is_suffix code ~suffix:";;" then code else code ^ ";;"
+let preformat_code code =
+  let is_repl_ready s = String.is_suffix s ~suffix:";;" in
+  let trimmed_code = String.strip code in
+  if is_repl_ready trimmed_code then
+    if String.mem trimmed_code '\n' then
```

Comment:
As far as I can tell `String.mem trimmed_code '\n'` is always evaluated. Therefore it should be assigned a binding and moved before the conditionals.